apps/web/src/middlewares/ip-address-middleware.ts

```diff
@@ -0,0 +1,17 @@
+import { NextFetchEvent, NextMiddleware, NextRequest, NextResponse } from 'next/server'
+import { shouldGeoBlock } from '@pancakeswap/utils/geoBlock'
+import { MiddlewareFactory } from './types'
+
+export const withUserIpHeaders: MiddlewareFactory = (next: NextMiddleware) => {
+  return async (request: NextRequest, _next: NextFetchEvent) => {
+    let ip = request.ip ?? request.headers.get('x-real-ip')
+    const forwardedFor = request.headers.get('x-forwarded-for')
+
+    if (!ip && forwardedFor) ip = forwardedFor.split(',').at(0) ?? 'Unknown'
+    if (!ip) return NextResponse.next()
+
+    request.headers.set('user-ip', ip)
```

Comment:
We can simply reassign req.ip. No need to add response header.

Also if we're adding custom header, it should start with x-, eg. x-user-ip. Since it's highly unlikely to be used outside our application